includes/wc-term-functions.php

```diff
@@ -434,21 +472,25 @@ function wc_terms_clauses( $clauses, $taxonomies, $args ) {
 		$clauses['fields']  .= ', tm.* ';
 	}
 
-	//query join
-	$clauses['join'] .= " LEFT JOIN {$wpdb->woocommerce_termmeta} AS tm ON (t.term_id = tm.woocommerce_term_id AND tm.meta_key = '". $meta_name ."') ";
+	// query join
+	if ( get_option( 'db_version' ) < 34370 ) {
+		$clauses['join'] .= " LEFT JOIN {$wpdb->woocommerce_termmeta} AS tm ON (t.term_id = tm.woocommerce_term_id AND tm.meta_key = '" . esc_sql( $meta_name ) . "') ";
+	} else {
+		$clauses['join'] .= " LEFT JOIN {$wpdb->termmeta} AS tm ON (t.term_id = tm.term_id AND tm.meta_key = '" . esc_sql( $meta_name ) . "') ";
+	}
 
 	// default to ASC
 	if ( ! isset( $args['menu_order'] ) || ! in_array( strtoupper($args['menu_order']), array('ASC', 'DESC')) ) {
 		$args['menu_order'] = 'ASC';
 	}
 
-	$order = "ORDER BY tm.meta_value+0 " . $args['menu_order'];
+	$order = "ORDER BY tm.meta_value+0 " . esc_sql( $args['menu_order'] );
```

Comment:
The usage of esc_sql is okay in lines 475-480, however the codex says esc_sql should not be used for "unquoted numeric values, field names, or SQL keywords" which this line and 490 does. See the notes section where it specifically mentions ORDER by. This line shouldn't be vulnerable in this instance since menu_order will only be ASC or DESC.. but should be able to just drop it.